### PR TITLE
Add universal integration tests

### PR DESCRIPTION
**Pull request by MattDelac**
**Which issue(s) this PR fixes**:

Fixes https://github.com/Shopify/feast-trino/issues/5

**What this PR does / why we need it**:
This PR is meant to add the Feast universal integration tests as part of our CI

It is dependent on https://github.com/Shopify/feast-trino/pull/27 as most of the fixes are there. This is why this PR is compared to `improvments_current_trino`

**Does this PR introduce a user-facing change?**:

```release-note
Add feast universal integration tests suite as part of our CI
```

### CODE REVIEW

File: .github/workflows/universal_tests.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+name: Run universal test suite
+
+on: [pull_request]
+
+jobs:
+  universal-tests:
+    runs-on: [ubuntu-latest]
+    steps:
```
Comment (paccorsi) [inline .github/workflows/universal_tests.yml:8]: Eventually, do we want to test the feast trino code on multiple version of Python?

File: Makefile [hunk 2]
```diff
@@ -19,10 +19,27 @@ lint:
...
+	mv ${ROOT_DIR}/tests_old ${ROOT_DIR}/tests
+
 build:
 	rm -rf dist/*
 	python setup.py sdist bdist_wheel
 
+install-feast-submodule:
+	cd ${ROOT_DIR}; git submodule add --force https://github.com/feast-dev/feast.git feast
+	cd ${ROOT_DIR}/feast; git fetch --all --tags
+	cd ${ROOT_DIR}/feast; git reset --hard tags/v0.15.1
```
[the remark this answers is not in the document]
Comment (MattDelac) [reply Makefile:39]: Yes good idea.

I wanted to have different matrix of
- python versions (3.7, 3.8, 3.9)
- feast version (0.15.*, 0.16.*, etc.)
- trino version (364, etc.)

I think it's captured in this issues https://github.com/Shopify/feast-trino/issues/21. I can create one issue for multiple feast version

File: .github/workflows/universal_tests.yml (hunk 1 above)
Comment (MattDelac) [reply .github/workflows/universal_tests.yml:8]: Yes https://github.com/Shopify/feast-trino/issues/21

File: .github/workflows/universal_tests.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
+name: Run universal test suite
+
+on: [pull_request]
```
[the remark this answers is not in the document]
Comment (MattDelac) [reply .github/workflows/universal_tests.yml:3]: I believe that it runs for any new commit on a opened PR not only when it gets opened
image

**Review by paccorsi — APPROVED** [review]

**Pull request merged**